evm/src/cpu/kernel/asm/bignum/ge.asm

```diff
@@ -0,0 +1,80 @@
+// Arithmetic on little-endian integers represented with 128-bit limbs.
+// All integers must be under a given length bound, and are padded with leading zeroes.
+
+// Returns a >= b.
+global ge_bignum:
+    // stack: len, a_start_loc, b_start_loc, retdest
+    SWAP1
+    // stack: a_start_loc, len, b_start_loc, retdest
+    DUP2
+    // stack: len, a_start_loc, len, b_start_loc, retdest
+    ADD
+    %decrement
+    // stack: a_end_loc, len, b_start_loc, retdest
+    SWAP2
+    // stack: b_start_loc, len, a_end_loc, retdest
+    DUP2
+    // stack: len, b_start_loc, len, a_end_loc, retdest
+    ADD
+    %decrement
+    // stack: b_end_loc, len, a_end_loc, retdest
+    %stack (b, l, a) -> (l, a, b)
+    // stack: len, a_end_loc, b_end_loc, retdest
+    %decrement
+ge_loop:
+    // stack: i, a_i_loc, b_i_loc, retdest
+    DUP3
+    DUP3
+    // stack: a_i_loc, b_i_loc, i, a_i_loc, b_i_loc, retdest
+    %mload_kernel_general
+    SWAP1
+    %mload_kernel_general
+    SWAP1
+    // stack: a[i], b[i], i, a_i_loc, b_i_loc, retdest
+    %stack (vals: 2) -> (vals, vals)
+    GT
+    %jumpi(greater)
+    // stack: a[i], b[i], i, a_i_loc, b_i_loc, retdest
+    LT
+    %jumpi(less)
+    // stack: i, a_i_loc, b_i_loc, retdest
+    DUP1
+    %eq_const(0)
```

Comment:
```suggestion
    ISZERO
```
Don't need to save `i` here as it's just thrown away after the jump anyway.

Reply:
Ah well don't I still need to `DUP` it, in case the loop continues?

Reply:
Oh yes, good catch! :)